hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/LocalContainerInfo.java

```diff
@@ -0,0 +1,165 @@
+
+    for (ContainerType containerType : ContainerType.values()) {
+      final Handler handler =
+          Handler.getHandlerForContainerType(
+              containerType,
+              conf,
+              datanodeUuid,
+              containerSet,
+              volumeSet,
+              metrics,
+              containerReplicaProto -> {
+              });
+      handler.setScmID(scmId);
+      handlers.put(containerType, handler);
+    }
+
+    ContainerController controller =
+        new ContainerController(containerSet, handlers);
+
+    Iterator<HddsVolume> volumeSetIterator = volumeSet.getVolumesList()
+        .iterator();
+
+    while (volumeSetIterator.hasNext()) {
+      HddsVolume volume = volumeSetIterator.next();
+      final ContainerReader reader =
+          new ContainerReader(volumeSet, volume, containerSet, conf);
+      reader.run();
+    }
+
+    if (containerId == 0L) {
+      Iterator<Container<?>> containerIt = controller.getContainers();
+      while (containerIt.hasNext()) {
+        Container container = containerIt.next();
+        outputContainer(container.getContainerData());
+      }
+    } else {
+      Container container = controller.getContainer(containerId);
+      if (container != null) {
+        outputContainer(container.getContainerData());
+      }
+    }
+
+    return null;
+  }
+
+  private void outputContainer(ContainerData data) throws IOException {
+    JsonObject result = new JsonObject();
+    result.addProperty("ContainerType", data.getContainerType().toString());
+    result.addProperty("ContainerID", data.getContainerID());
+    result.addProperty("LayoutVersion", data.getLayOutVersion().toString());
+    result.addProperty("ChunksPath", data.getChunksPath());
+    result.addProperty("State", data.getState().toString());
+    result.addProperty("MaxSize", data.getMaxSize());
+    result.addProperty("OriginPipelineId", data.getOriginPipelineId());
+    result.addProperty("OriginNodeId", data.getOriginNodeId());
+    result.addProperty("Volume", data.getVolume().toString());
+    result.addProperty("Checksum", data.getChecksum());
+    result.addProperty("DataScanTimestamp", data.getDataScanTimestamp());
+
+    switch (data.getContainerType()) {
+    case KeyValueContainer:
+      KeyValueContainerData kvData = (KeyValueContainerData)data;
+      result.addProperty("MetadataPath", kvData.getMetadataPath());
+      result.addProperty("ContainerDBType", kvData.getContainerDBType());
+      result.addProperty("DbFile", kvData.getDbFile().getPath());
+      result.addProperty("SchemaVersion", kvData.getSchemaVersion());
+      result.addProperty("DeleteTransactionId",
+          kvData.getDeleteTransactionId());
+      result.addProperty("BlockCommitSequenceId",
+          kvData.getBlockCommitSequenceId());
+    default:
+      break;
+    }
+
+    Gson gson2 = new GsonBuilder().setPrettyPrinting().create();
+    System.out.println(gson2.toJson(result));
```

Comment:
Would it be possible to use `JsonUtils.toJsonStringWithDefaultPrettyPrinter` to produce JSON output with less code?

Example:

https://github.com/apache/ozone/blob/85af50b84f2ecba81ff6b67095c4cd082da7b59f/hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/container/InfoSubcommand.java#L72

Reply:
@adoroszlai Actually I've considered this util, it seems that we may have to let the JsonUtils ignore some fields in `ContainerData` and `KeyValueContainerData` with a java annotation, which will add lines of code into the two class files.

Reply:
Hi @adoroszlai , now we are using this JsonUtils to dump more detailed output~